pkg/webhooks/ensurance/validating.go

```diff
@@ -0,0 +1,253 @@
+package ensurance
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	genericvalidation "k8s.io/apimachinery/pkg/api/validation"
+	metavalidation "k8s.io/apimachinery/pkg/apis/meta/v1/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/core"
+
+	ensuranceapi "github.com/gocrane/api/ensurance/v1alpha1"
+	"github.com/gocrane/crane/pkg/ensurance/collector"
+)
+
+type NepValidationAdmission struct {
+}
+
+type ActionValidationAdmission struct {
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (p *NepValidationAdmission) ValidateCreate(ctx context.Context, req runtime.Object) error {
+
+	nep, ok := req.(*ensuranceapi.NodeQOSEnsurancePolicy)
+	if !ok {
+		return fmt.Errorf("req can not convert to NodeQOSEnsurancePolicy")
+	}
+
+	allErrs := genericvalidation.ValidateObjectMeta(&nep.ObjectMeta, false, genericvalidation.NameIsDNS1035Label, field.NewPath("metadata"))
+
+	if nep.Spec.Selector != nil {
+		allErrs = append(allErrs, metavalidation.ValidateLabelSelector(nep.Spec.Selector, field.NewPath("spec").Child("selector"))...)
+	}
+
+	allErrs = append(allErrs, validateNodeQualityProbe(nep.Spec.NodeQualityProbe, field.NewPath("nodeQualityProbe"))...)
+
+	var httpGetEnable bool
+	if nep.Spec.NodeQualityProbe.HTTPGet != nil {
+		httpGetEnable = true
+	}
+	allErrs = append(allErrs, validateObjectiveEnsurances(nep.Spec.ObjectiveEnsurances, field.NewPath("objectiveEnsurances"), httpGetEnable)...)
+
+	if len(allErrs) != 0 {
+		return allErrs.ToAggregate()
+	}
+
+	return nil
+}
+
+func validateNodeQualityProbe(nodeProbe ensuranceapi.NodeQualityProbe, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+
+	if (nodeProbe.NodeLocalGet == nil) && (nodeProbe.HTTPGet == nil) {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("httpGet"), "", "The httpGet and nodeLocalGet must set one"))
```

Comment:
The httpGet and nodeLocalGet must set one->
`httpGet and nodeLocalGet cannot be empty at the same time.`